src/Spice86.Core/Emulator/OperatingSystem/DosProcessManager.cs

```diff
@@ -42,195 +64,509 @@ public DosProcessManager(IMemory memory, State state,
-        ms.WriteByte(0); // Null terminator for program path
-    
-        return ms.ToArray();
-    }
 
-    private void LoadComFile(byte[] com) {
-        ushort pspSegment = _pspTracker.GetCurrentPspSegment();
-        uint physicalLoadAddress = MemoryUtils.ToPhysicalAddress(pspSegment, DosProgramSegmentPrefix.PspSize);
-        _memory.LoadData(physicalLoadAddress, com);
+        string normalizedPath = programPath.Replace('/', '\\').ToUpperInvariant();
+        byte[] programPathBytes = Encoding.ASCII.GetBytes(normalizedPath);
+        ms.Write(programPathBytes, 0, programPathBytes.Length);
+        ms.WriteByte(0);
 
-        // Make SS, DS and ES point to the PSP
-        _state.DS = pspSegment;
-        _state.ES = pspSegment;
-        _state.SS = pspSegment;
-        _state.SP = 0xFFFE; // Standard COM file stack
-        SetEntryPoint(pspSegment, DosProgramSegmentPrefix.PspSize);
-        _state.InterruptFlag = true;
+        return ms.ToArray();
     }
 
-    private void LoadExeFile(DosExeFile exeFile, ushort pspSegment) {
-        if (_loggerService.IsEnabled(LogEventLevel.Verbose)) {
-            _loggerService.Verbose("Read header: {ReadHeader}", exeFile);
+    private void LoadExeFileInMemoryAndApplyRelocations(DosExeFile exeFile, ushort startSegment) {
+        uint physicalStartAddress = MemoryUtils.ToPhysicalAddress(startSegment, 0);
+        _memory.LoadData(physicalStartAddress, exeFile.ProgramImage, (int)exeFile.ProgramSize);
+        foreach (SegmentedAddress address in exeFile.RelocationTable) {
+            uint addressToEdit = MemoryUtils.ToPhysicalAddress(address.Segment, address.Offset)
+                + physicalStartAddress;
+            _memory.UInt16[addressToEdit] += startSegment;
         }
+    }
 
-        DosMemoryControlBlock? block = _memoryManager.ReserveSpaceForExe(exeFile, pspSegment);
-        if (block is null) {
-            throw new UnrecoverableException($"Failed to reserve space for EXE file at {pspSegment}");
+    private static void RestoreInterruptVector(byte vectorNumber, uint storedFarPointer,
+        InterruptVectorTable interruptVectorTable) {
+        if (storedFarPointer != 0) {
+            ushort offset = (ushort)(storedFarPointer & 0xFFFF);
+            ushort segment = (ushort)(storedFarPointer >> 16);
+            interruptVectorTable[vectorNumber] = new SegmentedAddress(segment, offset);
         }
-        ushort pspLoadSegment = block.DataBlockSegment;
-        // The program image is loaded immediately above the PSP, which is the start of
-        // the memory block that we just allocated.
-        // Jump over the PSP to get the EXE image segment.
-        ushort loadImageSegment = (ushort)(pspLoadSegment + DosProgramSegmentPrefix.PspSizeInParagraphs);
+    }
 
-        // Adjust image load segment when PSP and exe image gets splitted due to load into high memory
-        if (exeFile.MinAlloc == 0 && exeFile.MaxAlloc == 0) {
```

Comment:
This if (exeFile.MinAlloc == 0 && exeFile.MaxAlloc == 0) { is rare but it happens.

This if  should have been kept.